Stats: Adjust the latest post card layout and hide the most popular post

dognose24 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #70854 
## Proposed Changes

* Introduce the feature flag `stats/most-popular-post` to gate the intentionally hidden most popular post card.
* Complete the mobile layout for the latest post card appended to the all-time highlights.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Spin up this change with the Calypso Live link.
* Navigate to the `Stats` > `Insights` page.
* Ensure the `All-time highlights` and `Latest post summary` work as previously.
* Append the feature flag `?flags=stats/latest-post-stats` to the URL.
* Ensure there is the latest post card appended to the `All-time highlights` section and the `Latest post summary` is removed.
* Change the viewport to mobile.
* Ensure there is the mobile latest post card appended to the `All-time highlights` section.

##### PC
<img width="1271" alt="截圖 2023-02-09 下午5 11 34" src="https://user-images.githubusercontent.com/6869813/217768333-9c0ed039-bf21-4e4b-9e15-12945b3ee8ec.png">

##### Mobile
<img width="487" alt="截圖 2023-02-09 下午5 12 05" src="https://user-images.githubusercontent.com/6869813/217768386-becf7b3b-064a-4a85-944e-bcccc00899c6.png">

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

dognose24 committed:
Gate most popular post by feature flag

dognose24 committed:
Add LatestPostCard to the mobile layout

dognose24 committed:
Update title length to 60 for the LatestPostCard

keoshi commented:
Thanks for doing this, @dognose24!

Followed the instructions at the top and can confirm I do see the `Latest post` card at the top, but am still seeing the `Latest post summary` at the bottom. Am I missing anything?

<img width="415" alt="image" src="https://user-images.githubusercontent.com/390760/217864550-91e2358a-2ac9-432b-983e-3ed7792155e6.png">

Everything else works as expected and looks great!

Two more things worth mentioning in this context even though they're _not_ directly related to this PR:
- Created an issue in https://github.com/Automattic/wp-calypso/issues/73159 to propose a small change in the detailed view of the post that could also be applied here in the `Most popular/Latest post`.
- When the `dotPager` is used on mobile, I'm seeing a discrepancy between the entire card's background color (transparent/gray) and the background color on the inner content (white). This seems to happen because while there's a `background-color` defined for each inner `.card`, there's none for the `.dot-pager` element itself. Adding an image below to hopefully better illustrate the issue.

![image](https://user-images.githubusercontent.com/390760/217866787-612b16f1-78a1-496e-80c4-4641b4dbac87.png)

jeffgolenski commented:
Thanks @dognose24 aside from Filipe's comments above, can we make a couple of adjustments to the full width latest post card on large screens?

![Screen Shot 2023-02-09 at 11 34 43 AM](https://user-images.githubusercontent.com/214813/217878217-6e4ac214-187a-4f92-8313-e267d4a0ac7b.png)

- [ ] For the content area (title, published date, views, likes, comments), can we set a max width of `550px'? The space between this content area and the featured image can scale indefinitely. 


Current example:
![Screen Shot 2023-02-09 at 11 39 25 AM](https://user-images.githubusercontent.com/214813/217879599-6433b528-3f10-4df2-9c22-d485e28d79c2.png)


Proposed:
![Screen Shot 2023-02-09 at 11 40 34 AM](https://user-images.githubusercontent.com/214813/217879849-60e1148d-187a-43e9-beed-ab351840d1ce.png)

grzegorz-cp commented:
Great progress on the cards @dognose24. I noticed that both new feature flags are missing from the JSON config files. Could you add them as part of this or the next PR? It would help while I'm modernising other parts of the Insights page.

dognose24 committed:
Add feature flags to configs

dognose24 committed:
Adjust the space between card counts area columns

dognose24 committed:
Adjust the background color in line with content

dognose24 committed:
Add max-width to text content of PostStatsCard

dognose24 commented:
Thanks for all your design feedback! @jeffgolenski @keoshi 

> Created an issue in https://github.com/Automattic/wp-calypso/issues/73159 to propose a small change in the detailed view of the post that could also be applied here in the Most popular/Latest post.

Fixed in [68cb03f](https://github.com/Automattic/wp-calypso/pull/73144/commits/68cb03fa21957d40783e60ac4f611079bd304ae6).

> a discrepancy between the entire card's background color (transparent/gray) and the background color on the inner content (white)

Fixed in [e7a5ec9](https://github.com/Automattic/wp-calypso/pull/73144/commits/e7a5ec9abbf34590f8b04c1e2633787f80d3ee83).

> For the content area (title, published date, views, likes, comments), can we set a max width of `550px'?

Fixed in [9a2f355](https://github.com/Automattic/wp-calypso/pull/73144/commits/9a2f355ac7776dc2c0a21d1bc9dd51c24a24e934).

> It would help while I'm modernising other parts of the Insights page.

Sure! I have updated those configs in [f00b88f](https://github.com/Automattic/wp-calypso/pull/73144/commits/f00b88fd4a9d101b8172d5ad3d62fd47234b1483).

dognose24 committed:
Hide LatestPostSummary within the old grid layout

keoshi reviewed (approved):
Looks great to me, thanks so much for being so accommodating and making all the suggested changes! 🙌

jsnmoon reviewed (approved):
Let's ship and iterate since this is hidden behind a feature flag (or two).

One thing we should consider in a follow-up PR: I noticed that the thumbnail column failed to collapse even when there was no thumbnail to display.

<img width="1256" alt="image" src="https://user-images.githubusercontent.com/4044428/218195105-e16b798b-6bab-4dcd-ba2d-d0aec43b0d11.png">

<img width="1251" alt="image" src="https://user-images.githubusercontent.com/4044428/218195345-23fdd14c-e1de-462f-9431-38c5ab41ab8e.png">

jeffgolenski commented:
Y'all are great. Thanks to everyone for the quick iterations!
